frontend/src/pages/hardwareProfiles/utils.ts

```diff
@@ -1,4 +1,229 @@
+        parentWarningMessage = `Minimum allowed ${
+          identifier.resourceType ?? identifier.displayName
+        } cannot be negative.`;
+        continue;
+      }
+      if (identifier.maxCount.toString().at(0) === '-') {
+        parentWarningMessage = `Maximum allowed ${
+          identifier.resourceType ?? identifier.displayName
+        } cannot be negative.`;
+        continue;
+      }
+      if (identifier.defaultCount.toString().at(0) === '-') {
+        parentWarningMessage = `Default count for ${
+          identifier.resourceType ?? identifier.displayName
+        } cannot be negative.`;
+        continue;
+      }
+      const [minCount] = splitValueUnit(
+        identifier.minCount.toString(),
+        determineUnit(identifier),
+        true,
+      );
+      const [maxCount] = splitValueUnit(
+        identifier.maxCount.toString(),
+        determineUnit(identifier),
+        true,
+      );
+      const [defaultCount] = splitValueUnit(
+        identifier.defaultCount.toString(),
+        determineUnit(identifier),
+        true,
+      );
+      if (!Number.isInteger(minCount)) {
+        parentWarningMessage = `Minimum count for ${
+          identifier.resourceType ?? identifier.displayName
+        } cannot be a decimal.`;
+      }
+      if (!Number.isInteger(maxCount)) {
+        parentWarningMessage = `Maximum count for ${
+          identifier.resourceType ?? identifier.displayName
+        } cannot be a decimal.`;
+      }
+      if (!Number.isInteger(defaultCount)) {
+        parentWarningMessage = `Default count for ${
+          identifier.resourceType ?? identifier.displayName
+        } cannot be a decimal.`;
+      }
+      if (minCount > maxCount) {
+        parentWarningMessage = `Minimum allowed ${
+          identifier.resourceType ?? identifier.displayName
+        } cannot exceed maximum allowed ${identifier.resourceType ?? identifier.displayName}.`;
+      } else if (defaultCount < minCount || defaultCount > maxCount) {
+        parentWarningMessage = `The default count for ${
+          identifier.resourceType ?? identifier.displayName
+        } must be between the minimum allowed ${
+          identifier.resourceType ?? identifier.displayName
+        } and maximum allowed ${identifier.resourceType ?? identifier.displayName}.`;
+      }
+    } catch (e) {
+      parentWarningMessage = `The resource count for ${
+        identifier.resourceType ?? identifier.displayName
+      } has an invalid unit.`;
+    }
+  }
+
+  return parentWarningMessage
+    ? {
+        title: `${!hasCPUandMemory(identifiers) ? 'Incomplete' : 'Invalid'} ${
+          isDefaultProfile === true ? 'default hardware' : 'hardware'
+        } profile`,
+        message: `${parentWarningMessage} ${
+          isDefaultProfile
+            ? HARDWARE_PROFILES_DEFAULT_WARNING_MESSAGE
+            : 'Edit the profile to make the profile valid.'
+        }`,
+      }
+    : undefined;
+};
```

Comment:
this function should basically validate the hardware profile. It looks like you are returning only one error becuase parentWarningMessage keeps getting overwritten.

I think this function should return an array of all the warnings with the profile. so...

```tsx
enum HardwareProfileWarningType {
   HARDWARE_PROFILES_DEFAULT_WARNING,
   HARDWARE_PROFILES_MISSING_CPU_MEMORY,
   OTHER
}
   
type HardwareProfileWarning = {
   title: string,
   message: string
   type: HardwareProfileWarningType
}

export const validateProfileWarning = ( hardwareProfile: HardwareProfileKind): HardwareProfileWarning[] => {
  // same func body but instead add to this array and return that
}